jcl/src/java.base/share/classes/openj9/internal/criu/security/CRIUConfigurator.java

```diff
@@ -42,7 +46,9 @@ public final class CRIUConfigurator {
 	private static final HashMap<String, String> oldProviders = new HashMap<>();
 	/** Tracing for CRIUSEC. */
 	private static final boolean debug = Boolean.parseBoolean(
-			GetPropertyAction.privilegedGetProperty("enable.j9internal.checkpoint.security.api.debug", "false"));
+		 GetPropertyAction.privilegedGetProperty("enable.j9internal.checkpoint.security.api.debug", "false"));
+	private static HashMap<String, Set<String>> cachedAlgorithms = new HashMap<>();
+	private static boolean isCacheEligible = false;
```

Comment:
The type is now `Map` and the `cachedAlgorithms` variable is marked final.

For the `isCacheEligible` update I do find it easier to read code where the value is initialized explicitly to false. Yes the default value is false for a boolean although I would doubt that we are double initalizing this value to false. Does the JVM not optimize this out to avoid any cost in setting it to false twice? I am unsure there. This all being said I removed the `= false` so the reader will just have to know its false.

Reply:
No, `javac` produces instructions to (redundantly) set the field. I don't think there's any optimization that eliminates that work.